internal/vault/credentials/approle.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package credentials
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	secretsv1alpha1 "github.com/hashicorp/vault-secrets-operator/api/v1alpha1"
+)
+
```

Comment:
would be good to declare that `AppRoleCredentialProvider` implements `CredentialProvider` interface e.g:  https://github.com/hashicorp/vault-secrets-operator/blob/2081cc71f0cc9fee11a5e3a7e2b8bb1939ef8697/internal/vault/credentials/kubernetes.go#L17